src/sentry/static/sentry/app/views/performance/landing.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import {Location} from 'history';
+import * as ReactRouter from 'react-router';
+
+import {t} from 'app/locale';
+import {Organization} from 'app/types';
+import withOrganization from 'app/utils/withOrganization';
+import SentryDocumentTitle from 'app/components/sentryDocumentTitle';
+import GlobalSelectionHeader from 'app/components/organizations/globalSelectionHeader';
+import {PageContent} from 'app/styles/organization';
+import NoProjectMessage from 'app/components/noProjectMessage';
+import Alert from 'app/components/alert';
+import EventView from 'app/views/eventsV2/eventView';
+import {getUtcToLocalDateObject} from 'app/utils/dates';
+import {getParams} from 'app/components/organizations/globalSelectionHeader/getParams';
+import {StyledPageHeader} from 'app/views/eventsV2/landing';
+
+import {generatePerformanceQuery, DEFAULT_STATS_PERIOD} from './data';
+import Table from './table';
+import Charts from './charts/index';
+
+type Props = {
+  organization: Organization;
+  location: Location;
+  router: ReactRouter.InjectedRouter;
+};
+
+type State = {
+  eventView: EventView;
+  error: string | undefined;
+};
+
+function generatePerformanceEventView(location: Location): EventView {
+  return EventView.fromNewQueryWithLocation(generatePerformanceQuery(location), location);
+}
+
+class PerformanceLanding extends React.Component<Props, State> {
+  static getDerivedStateFromProps(nextProps: Props, prevState: State): State {
+    return {...prevState, eventView: generatePerformanceEventView(nextProps.location)};
+  }
+
+  state = {
+    eventView: generatePerformanceEventView(this.props.location),
+    error: undefined,
+  };
+
+  renderError = () => {
+    const {error} = this.state;
+
+    if (!error) {
+      return null;
+    }
+
+    return (
+      <Alert type="error" icon="icon-circle-exclamation">
+        {error}
+      </Alert>
+    );
+  };
+
+  setError = (error: string | undefined) => {
+    this.setState({error});
+  };
+
+  generateGlobalSelection = () => {
```

Comment:
Having another implementation of this increases the chances that we will have new global selection bugs.